### PR TITLE
Make method annotations more strict

### PR DESCRIPTION
**Pull request by Harfusha**
https://github.com/cakephp/bake/issues/1002

### CODE REVIEW

**Comment by dereuromark** [comment]
The original issue reported was

```
> newEntity() parameter #1 $data with no value type specified in iterable type array.
```
None of the concrete entity changes is addressing it.
The opposite: this is usually rather creating more problems for App level.

Can we just fix the mixed in iterable instead?

**Comment by dereuromark** [comment]
I would expect
```
iterable<EntityInterface>
```
For input args (param) annotation 

And concrete entity for output (return) annotation

**Comment by Harfusha** [comment]
Oh, so you want to implement the concete entities in IdeHelper?
Going to fix the issues.

**Review by dereuromark — DISMISSED** [review]
Looks good

File: src/View/Helper/DocBlockHelper.php [hunk 1]
```diff
@@ -290,20 +290,23 @@ public function buildTableAnnotations(
...
-        $annotations[] = "@method iterable<\\{$namespace}\\Model\\Entity\\{$entity}>|\Cake\Datasource\ResultSetInterface<\\{$namespace}\\Model\\Entity\\{$entity}>|false deleteMany(iterable \$entities, array \$options = [])";
-        $annotations[] = "@method iterable<\\{$namespace}\\Model\\Entity\\{$entity}>|\Cake\Datasource\ResultSetInterface<\\{$namespace}\\Model\\Entity\\{$entity}> deleteManyOrFail(iterable \$entities, array \$options = [])";
+        $annotations[] = "@method $entityPath newEmptyEntity()";
+        $annotations[] = "@method $entityPath newEntity(array<mixed> \$data, array<string, mixed> \$options = [])";
+        $annotations[] = "@method array<$entityPath> newEntities(array<mixed> \$data, array<string, mixed> \$options = [])";
+        $annotations[] = "@method $entityPath get(mixed \$primaryKey, array<mixed>|string \$finder = 'all', \\Psr\\SimpleCache\\CacheInterface|string|null \$cache = null, \Closure|string|null \$cacheKey = null, mixed ...\$args)";
```
Comment (ADmad) [inline src/View/Helper/DocBlockHelper.php]: Passing an array for the `$finder` arg is deprecated and kept just for BC, so I would suggest just going with `string $finder`

**Pull request closed without merging**